### PR TITLE
[deckhouse] one source of truth for deckhouse registry (stage #1)

### PR DESCRIPTION
**Pull request by fl64**
## Description
Removed functionality to retrieve registry data (`registry` and `path`) from the deckhouse image for internal values, now hook get it form docker-registry secret.
The flant-integration module retrieves license information from internal variables, not from the mounted secret deckhouse-registry.
All registry settings are moved to `global.moduleImages.registry.*`:
* global.moduleImages.registry --> global.moduleImages.registry.base
* global.moduleImages.registryDockercfg --> global.moduleImages.registry.dockercfg
* global.moduleImages.registryScheme --> global.moduleImages.registry.scheme
* global.moduleImages.registryCA --> global.moduleImages.registry.CA
* global.moduleImages.registryAddress --> global.moduleImages.registry.address
* global.moduleImages.registryPath --> global.moduleImages.registry.path

## Why do we need it, and what problem does it solve?
This PR closes part of the issue (https://github.com/deckhouse/deckhouse/issues/714)

## What is the expected result?
All registry settings are located at`global.moduleImages.registry` key.

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: chore
summary: Changed deckhouse_registry hook to get registry data form secret docker-registry. The global values of the registry is refactored for all modules.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @fl64. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — CHANGES_REQUESTED** [review]
one thing and need rebase

File: global-hooks/discovery/deckhouse_registry.go [hunk 1]
```diff
@@ -76,107 +56,51 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
...
-
 	if len(registryConfSnap) == 0 {
-		return fmt.Errorf("not found deckhouse registry conf secret")
+		return fmt.Errorf("not found 'deckhouse-registry' secret")
 	}
 
 	registrySecretRaw := registryConfSnap[0].(*registrySecret)
 	// yes, we store base64 encoded string but in secret object store decoded data
 	// In values we store base64-encoded docker config because in this form it is applied in other places.
 	registryConfEncoded := base64.StdEncoding.EncodeToString(registrySecretRaw.RegistryDockercfg)
```
[the remark this answers is not in the document]
Comment (fl64) [reply global-hooks/discovery/deckhouse_registry.go:98]: The `dockercfg` check already exists in the filter, so I added the `address` field check there. But I'm not sure about the path.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @fl64. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by name212 — DISMISSED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @fl64. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — DISMISSED** [review]

**Review by yalosev — DISMISSED** [review]

**Review by apolovov — CHANGES_REQUESTED** [review]
LGTM, but it would be nice to rename all templates to registry-secret.yaml

**Comment by shvgn** [comment]
Regarding flant-integation changes, it look good to me. 

Regarding changelog block, there should be improvements. 

1. Valid YAML
2. I think flant-integration it doesn't deserve a special mention among all other modules.

Here is my suggestion for the summary, omitting internal detail

>  Changed `deckhouse_registry` hook to get registry data form secret `docker-registry`. The global values of the registry is refactored for all modules.

**Review by shvgn — DISMISSED** [review]

**Review by apolovov — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]

**Comment by kvaps** [comment]
@RomanenkoDenys could you confirm if this is not affecting this script, please:

https://github.com/deckhouse/deckhouse/blob/e5bd9d7e2818ee7610d2205e340e644f9f7d4993/candi/bashible/bashible.sh.tpl#L121-L126

**Review by kvaps — COMMENTED** [review]
@fl64 docs are not updated https://github.com/deckhouse/deckhouse/blob/e5bd9d7e2818ee7610d2205e340e644f9f7d4993/docs/documentation/pages/DECKHOUSE-FAQ.md?plain=1#L122-L123

**Comment by fl64** [comment]
> @fl64 docs are not updated
> 
> https://github.com/deckhouse/deckhouse/blob/e5bd9d7e2818ee7610d2205e340e644f9f7d4993/docs/documentation/pages/DECKHOUSE-FAQ.md?plain=1#L122-L123
@kvaps 
I'm not sure we need to update this document, because these settings refer to configs for dhctl, and my changes concerned internal variables.
@name212 Can you please tell me if I'm right?

**Pull request merged**